pkg/app/piped/cloudprovider/kubernetes/helm.go

```diff
@@ -199,3 +212,70 @@ func (c *Helm) TemplateRemoteChart(ctx context.Context, appName, appDir, namespa
 	}
 	return executor()
 }
+
+// verifyHelmValueFilePath verifies if the path of the values file references
+// a remote URL or inside the path where the application configuration file (i.e. *.pipecd.yaml) is located.
+func verifyHelmValueFilePath(appDir, valueFilePath string) error {
+	url, err := url.Parse(valueFilePath)
+	if err == nil && url.Scheme != "" {
+		for _, s := range allowedURLSchemes {
+			if strings.EqualFold(url.Scheme, s) {
+				return nil
+			}
+		}
+
+		return fmt.Errorf("scheme %s is not allowed to load values file", url.Scheme)
+	}
+
+	// absAppDir is a path where ".pipecd.yaml" is located.
+	absAppDir, err := filepath.Abs(appDir)
+	if err != nil {
+		return err
+	}
+
+	// valueFilePath is a path where non-default Helm values file is located.
+	if !filepath.IsAbs(valueFilePath) {
+		valueFilePath = filepath.Join(absAppDir, valueFilePath)
+	}
+
+	valueFilePath, err = resolveSymlink(valueFilePath)
+	if err != nil {
+		return err
+	}
+
+	// convert to absolute path again in case symlink is resolved.
+	if !filepath.IsAbs(valueFilePath) {
+		valueFilePath = filepath.Join(absAppDir, valueFilePath)
+	}
```

Comment:
how about checking whether a valueFilePath is symlink or not first and only resolveSymlink in case it's a symlink. Inside resolveSymlink function, we can call this filepath to ensure only the absolute path is returned. If we write code that way, can avoid `convert to absolute path again in case ...`